pydantic_factories/value_generators/constrained_number.py

```diff
@@ -18,7 +19,7 @@ def get_increment(t_type: Type[T]) -> T:
     Returns:
         An increment T.
     """
-    values: Dict[Any, Any] = {int: 1, float: 0.001, Decimal: Decimal("0.001")}
+    values: Dict[Any, Any] = {int: 1, float: sys.float_info.epsilon, Decimal: Decimal("0.001")}
```

Comment:
Updated here float increment with proper value of the (`sys.float_info.epsilon` is the smallest float increment possible, `1.0 + sys.float_info.epsilon/2 == 1` yields `True`)

Decimal increment is tricky. On one hand decimals can have infinite precision. On the other hand we have `decimal_places` argument for [`condecimal`](https://pydantic-docs.helpmanual.io/usage/types/#arguments-to-condecimal).  

We will have to addressed this.